node/nodeRunner.go

```diff
@@ -368,15 +382,51 @@ func (nr *nodeRunner) createApiFacade(currentNode *Node, gasScheduleNotifier cor
 
 	ef, err := facade.NewNodeFacade(argNodeFacade)
 	if err != nil {
-		return nil, fmt.Errorf("%w while creating NodeFacade", err)
+		return nil, nil, fmt.Errorf("%w while creating NodeFacade", err)
 	}
 
 	ef.SetSyncer(currentNode.coreComponents.SyncTimer())
 	ef.SetTpsBenchmark(currentNode.statusComponents.TpsBenchmark())
 
+	err = closableHttpServer.Close()
```

Comment:
why do we need Close calling here?

Reply:
Because of the `HttpServerCloser` interface: you control the starting and the closing from outside, so before upgrading, we should close this one (othwewise => `address already in use`)